torch/_dynamo/variables/torch_function.py

```diff
@@ -160,11 +213,24 @@ def call_method(
         # of `call_method`.
         if tx.output.torch_function_enabled:
             import torch
-            from .builder import SourcelessBuilder
+            from .builder import SourcelessBuilder, VariableBuilder
+
+            if _is_attr_overidden(tx, self, name):
+                unimplemented(
+                    f"Calling overidden method {name} on a tensor"
+                    " subclass with a __torch_function__ override is not supported"
+                )
 
             # [Note: __torch_function__] Currently we only support methods that are defined on tensor
             # we will graph break in other cases this will need a bigger overhaul of extracting methods/comparing them for equality
-            func_var = SourcelessBuilder()(tx, getattr(torch.Tensor, name))
+            # We've established with the above check that the method is not overridden, so we guard that the method is the same
+            # as the impl defined on tensor and retrieve it
```

Comment:
Hmm... this feels like over-guarding to me. This seems to imply that we will generate a guard for EVERY function access on the subclass to guard against the user post-facto monkeypatching their class to start overriding a method. It would be much better to do one of the version dict things we were doing for nn.Module to just test that, yes, the user hasn't modified their user defined class in the meantime.